pkg/engine/datasource/kafka_datasource/config.go

```diff
@@ -0,0 +1,124 @@
+package kafka_datasource
+
+import (
+	"fmt"
+
+	"github.com/Shopify/sarama"
+)
+
+const (
+	IsolationLevelReadUncommitted = "ReadUncommitted"
+	IsolationLevelReadCommitted   = "ReadCommitted"
+)
+
+const DefaultIsolationLevel = IsolationLevelReadUncommitted
+
+const (
+	BalanceStrategyRange      = "BalanceStrategyRange"
+	BalanceStrategySticky     = "BalanceStrategySticky"
+	BalanceStrategyRoundRobin = "BalanceStrategyRoundRobin"
+)
+
+const DefaultBalanceStrategy = BalanceStrategyRange
+
+var (
+	DefaultKafkaVersion          = "V1_0_0_0"
+	SaramaSupportedKafkaVersions = map[string]sarama.KafkaVersion{
+		"V0_10_2_0": sarama.V0_10_2_0,
+		"V0_10_2_1": sarama.V0_10_2_1,
+		"V0_11_0_0": sarama.V0_11_0_0,
+		"V0_11_0_1": sarama.V0_11_0_1,
+		"V0_11_0_2": sarama.V0_11_0_2,
+		"V1_0_0_0":  sarama.V1_0_0_0,
+		"V1_1_0_0":  sarama.V1_1_0_0,
+		"V1_1_1_0":  sarama.V1_1_1_0,
+		"V2_0_0_0":  sarama.V2_0_0_0,
+		"V2_0_1_0":  sarama.V2_0_1_0,
+		"V2_1_0_0":  sarama.V2_1_0_0,
+		"V2_2_0_0":  sarama.V2_2_0_0,
+		"V2_3_0_0":  sarama.V2_3_0_0,
+		"V2_4_0_0":  sarama.V2_4_0_0,
+		"V2_5_0_0":  sarama.V2_5_0_0,
+		"V2_6_0_0":  sarama.V2_6_0_0,
+		"V2_7_0_0":  sarama.V2_7_0_0,
+		"V2_8_0_0":  sarama.V2_8_0_0,
+	}
+)
+
+type GraphQLSubscriptionOptions struct {
+	BrokerAddr           string `json:"broker_addr"`
+	Topic                string `json:"topic"`
+	GroupID              string `json:"group_id"`
+	ClientID             string `json:"client_id"`
+	KafkaVersion         string `json:"kafka_version"`
+	StartConsumingLatest bool   `json:"start_consuming_latest"`
+	BalanceStrategy      string `json:"balance_strategy"`
+	IsolationLevel       string `json:"isolation_level"`
+	startedCallback      func()
+}
+
+func (g *GraphQLSubscriptionOptions) Sanitize() {
+	if g.KafkaVersion == "" {
+		g.KafkaVersion = DefaultKafkaVersion
+	}
+
+	// Strategy for allocating topic partitions to members (default BalanceStrategyRange)
+	if g.BalanceStrategy == "" {
+		g.BalanceStrategy = DefaultBalanceStrategy
+	}
+
+	if g.IsolationLevel == "" {
+		g.IsolationLevel = DefaultIsolationLevel
+	}
+}
+
+func (g *GraphQLSubscriptionOptions) Validate() error {
```

Comment:
Do you think it would be worth creating an array of the strings (possibly change them to bytes) and use a for loop? Something like
`
options := []string{BrokerAddr, Topic, GroupId...}
for _, option:= range options {
    if option == "" {
        return fmt.Errorf("%s cannot be empty", option[5:len(option)-2]`
    }
}

Reply:
I tend to stay away from reflections, it can get messy (plus it's usually expensive)
If you really want to avoid ifs i'd recommend
```
switch "" {
	case g.BrokerAddr:
		return fmt.Errorf("broker_addr cannot be empty")
	case g.Topic:
		return fmt.Errorf("topic cannot be empty")
	case g.GroupID:
		return fmt.Errorf("group_id cannot be empty")
	case g.ClientID:
		return fmt.Errorf("client_id cannot be empty")
	}
```

Reply:
I prefer to use a switch-case statement.